File: Python/Q51.py
# Q51.py - Program to Swap Numbers in Cyclic Order Using Call by Reference
def swap_cyclic(a, b, c):
    return b, c, a

a = 1; b = 2; c = 3
a, b, c = swap_cyclic(a, b, c)
print(f"After cyclic swap: a={a}, b={b}, c={c}")

File: C/Q51.c
#include <stdio.h>

// Function to swap numbers in cyclic order
void cyclicSwap(int *a, int *b, int *c) {
    int temp = *a;
    *a = *c;
    *c = *b;
    *b = temp;
}

int main() {
    int x = 1, y = 2, z = 3;

    printf("Before swapping:\n");
    printf("x = %d, y = %d, z = %d\n", x, y, z);

    cyclicSwap(&x, &y, &z);

    printf("After cyclic swapping:\n");
    printf("x = %d, y = %d, z = %d\n", x, y, z);

    return 0;
}
